webhook/resourcesemantics/validation/validation_admit.go

```diff
@@ -77,8 +77,16 @@ func (ac *reconciler) Admit(ctx context.Context, request *admissionv1.AdmissionR
 		return webhook.MakeErrorStatus("decoding request failed: %v", err)
 	}
 
-	if err := validate(ctx, resource, request); err != nil {
-		return webhook.MakeErrorStatus("validation failed: %v", err)
+	errors, warnings := validate(ctx, resource, request)
+	if warnings != nil {
+		// If there were warnings, then keep processing things, but augment
+		// whatever AdmissionResponse we send with the warnings.
+		defer func() {
```

Comment:
Just curious why this is defer instead of just resp.Warnings = ...

Reply:
It's mostly for plumbing.  This way of writing it catches whatever we return and sets `Warnings` on it.

If we were just to set it this way, then all of the return paths below here actually would overwrite it since `return webhook.MakeErrorStatus` is really the equivalent of:
```go
// Overwrite resp entirely, including Warnings.
resp = webhook.MakeErrorStatus(...)
return resp
```

The named return variable and defer let us get a crack at the actual returned value, and alter it.

Reply:
It might be worth a comment that the regular form Ville thought of doesn't work here because of the function returns below.

I also wonder whether `validate` should return a list of api.FieldErrors and this function separates them, but that seems like six of one and a half-dozen of the other.

Reply:
Added a comment about this.

For the return value, I did it this way because of the `errMissingNewObject` path, but we could always make this a field error (and/or do this refactor later, since this is private).